clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -188,6 +188,12 @@ void run(long now) {
                 notReadyTimeout = Math.min(notReadyTimeout, this.client.connectionDelay(node, now));
             }
         }
+		
+		// expire batch before draining to get some effective RecordBatches
```

Comment:
In current design (KIP-19), expiration of a batch in the accumulator is to avoid holding the batch forever in the accumulator when the partitions has no leader. If the batch can be drained, that means it can still make progress. In that case, we probably don't want to expire that batch.